### ISSUE TITLE
Support for MapStruct SPI

### ISSUE THREAD

**Issue by coney** [mapstruct/mapstruct-idea#91]
My DTOs are auto generated. each field is generated with an setter(setXXX) and chained accessor(withXXX). I wrote a customized `AccessorNamingStrategy` SPI to exclude those methods start with `with` and added the SPI to my project's preprocessor.

My SPI has successfully enlimited warnings of `withXXX` accessors. but in my IDEA, it still promotes the hits for adding `withXXX` mappings:
![image](https://user-images.githubusercontent.com/3133778/153820569-f0ef674a-9b08-4c37-8b28-1187d8275475.png)

**Comment by 91wzhang** [comment]
Having same issue when trying to use the protobuf SPI

**Comment by filiphr** [comment]
@coney This ia a tricky one to be honest. There is no way for us to invoke the SPI from within the plugin. What we could offer perhaps is a way for people to somehow configure how certain methods need to be ignored.

@91wzhang supporting the protobuf SPI is perhaps that we need to look into independently. I would suggest creating a separate issue for that

**Comment by coney** [comment]
> @coney This ia a tricky one to be honest. There is no way for us to invoke the SPI from within the plugin. What we could offer perhaps is a way for people to somehow configure how certain methods need to be ignored.
> 
> @91wzhang supporting the protobuf SPI is perhaps that we need to look into independently. I would suggest creating a separate issue for that

Great, maybe a simple configurable regex pattern is enough for me

**Comment by filiphr** [comment]
>Great, maybe a simple configurable regex pattern is enough for my scenario

That is an interesting idea @coney. We can look into that one.

**Comment by filiphr** [comment]
@coney would you be interested in providing this functionality to the plugin?

**Comment by jomu78** [comment]
@filiphr : I gave it a first shot - see https://github.com/mapstruct/mapstruct-idea/pull/150

**Comment by filiphr** [comment]
Thanks @jomu78. I've added a response on the PR in https://github.com/mapstruct/mapstruct-idea/pull/150#issuecomment-1712785314

**Comment by filiphr** [comment]
@kruijs-bbraun I don't think that anyone is working on a solution for this. My old comment (https://github.com/mapstruct/mapstruct-idea/issues/91#issuecomment-1184806178) still stands.
